articles/diagnose-diabetes-with-svm/index.md

```diff
@@ -0,0 +1,342 @@
+**First Rows of the dataset**
+
+![Last Rows of the dataset](/engineering-education/diagnose-diabetes-with-svm/last-rows-of-dataset.jpg)
+**Last Rows of the dataset**
+
+### Feature Extraction
+We separate the features and the target variable. We have eight features.
+
+```python
+# Extract Features
+X = dataset.iloc[:, :8]
+X.head()
+```
+
+![Dataset Features](/engineering-education/diagnose-diabetes-with-svm/dataset-features.jpg)
+**Dataset Features**
+
+Our target variable is the outcome column. The value 1 represents patients with diabetes, while 0 represents patients without diabetes.
+
+```python
+# Extract Class Labels
+y = dataset["Outcome"]
+y.head()
+```
+
+![Class Labels of the Dataset](/engineering-education/diagnose-diabetes-with-svm/class-labels.jpg)
+**Class Labels**
+
+### Split Dataset Into Training and Test Set
+We split our dataset into the training and test set. We use 75% of our dataset for training the model, and we use the remaining 25% for testing the model after training.
+
+```python
+# Split Dataset
+X_train, X_test, y_train, y_test = train_test_split(X, y, train_size=0.75, random_state=0)
+    
+print(X_train.shape)
+```
+
+![Shape of X_train](/engineering-education/diagnose-diabetes-with-svm/xtrain-shape.jpg)
+
+```python
+print(y_train.shape)
+```
+
+![Shape of y_train](/engineering-education/diagnose-diabetes-with-svm/ytrain-shape.jpg)
+
+```python
+print(X_test.shape)
+```
+
+![Shape of X_test](/engineering-education/diagnose-diabetes-with-svm/xtest-shape.jpg)
+
+```python
+print(y_test.shape)
+```
+
+![Shape of y_test](/engineering-education/diagnose-diabetes-with-svm/ytest-shape.jpg)
+
+We can see the amount of data that we will use for training and testing.
+
+```python
+X_train.head()
+```
+
+![Training Set](/engineering-education/diagnose-diabetes-with-svm/training-set.jpg)
+**Training Set**
+
+We need to normalize the features in our training set. Normalizing adjusts each column in our dataset to have a mean of 0 and a standard deviation of 1. It will make the training process faster.
+
+```python
+# Normalize Features
+scaler = StandardScaler()
+scaler.fit(X_train)
+X_train = scaler.transform(X_train)
+```
+
+Let us view the first five rows of our training set after normalization.
+
+```python
+# View first 5 rows
+X_train[:5, :]
+```
+
+![Normalized Training Set](/engineering-education/diagnose-diabetes-with-svm/normalized-train-set.jpg)
+**Normalized Training Set**
+
+### Creating the SVM Model
+The Sci-kit Learn library has four SVM kernels. We have the linear, poly, rbf, and sigmoid kernels. We do not know which of these kernels will give us a better decision boundary. So we iterate through the kernels and see which one gives us the best decision boundary for the dataset. The decision boundary separates the positive class and the negative class. It could be linear or non-linear. The polynomial and RBF kernels are suitable when the classes are not linearly separable.
```

Comment:
> decision boundary

Please explain what a decision boundary is with a quick 1-2 line definition. It'll help someone who is relatively new to ML to follow quickly

Reply:
Explanation with an image example has been added